CI with GH token validates LICENSE-3rdParty.csv

sdavtaker opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the project's contributing guide
     - 👷‍♀️ Create small PRs.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.
-->

## What type of PR is this? (check all applicable)

- [x] Feature / Major Change / Refactor / Optimization
- [ ] Bug Fix
- [ ] Documentation Update

## Description

- In integration CI GITHUB_TOKEN is used to avoid throttling of tests.
- In integration CI, dd-license-attribution LICENSE-3rdparty.csv is regenerated and compared to the one that is in the repository, if they are not equal it fails CI.

## Related tickets & Documents

<!--
For new features or major changes, we follow a documented RFC process (Check our CONTRIBUTING guidelines).
We cannot accept new features or major changes without a linked approved RFC.

For pull requests that relate or close an issue, please include them below.
-->

- Approved RFC (for feature/major changes) #
- Related Issue #
- Closes #

## How to reproduce and testing

<!--
For bug fixes, please describe how you reproduced the issue, what environments this was tested on (architecture, OS, etc.), and how the PR solves the issue.
-->

sdavtaker committed:
Using default GITHUB_TOKEN for test and validating output

arapulido commented on .github/workflows/integration-test.yml:
```diff
@@ -36,9 +36,19 @@ jobs:
+          if ! cmp -s ddla-output.txt LICENSE-3rdparty.csv; then
+            echo "Error: ddla-output.txt differs from LICENSE-3rdparty.csv"
+            echo "Diff between files:"
+            diff ddla-output.txt LICENSE-3rdparty.csv || true
```
Can we `diff -u --color` instead, so it is easier to read?

sdavtaker replied:
```
+"beartype","https://github.com/beartype/beartype","['\n        FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\n        AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM', '\n        OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE\n        SOFTWARE.', ' ARISING FROM', ' DAMAGES OR OTHER\n        LIABILITY', ' EXPRESS OR\n        IMPLIED', ' INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY', ' TORT OR OTHERWISE', ' WHETHER IN AN ACTION OF CONTRACT', ' WITHOUT WARRANTY OF ANY KIND', ' and to permit persons to whom the Software is\n        furnished to do so', ' and/or sell\n        copies of the Software', ' copy', ' distribute', ' free of charge', ' including without limitation the rights\n        to use', ' merge', ' modify', ' publish', ' subject to the following conditions:\n        \n        The above copyright notice and this permission notice shall be included in all\n        copies or substantial portions of the Software.\n     
```
doesn't have closing ]" for some reason, looking into it.

arapulido reviewed (changes requested):
If I run the command in my laptop, both in `main` and your branch, I get the same current value, so my guess is that there is a slight difference between dependencies in Linux and MacOS. 

We should generate the file using Linux, so it is reproducible, but we need to add some extra overrides. Also I have some questions about the diff. Commenting that one, line by line:

```
< "beartype","https://github.com/beartype/beartype","['\n        FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\n        AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM', '\n        OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE\n        SOFTWARE.', ' ARISING FROM', ' DAMAGES OR OTHER\n        LIABILITY', ' EXPRESS OR\n        IMPLIED', ' INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY', ' TORT OR OTHERWISE', ' WHETHER IN AN ACTION OF CONTRACT', ' WITHOUT WARRANTY OF ANY KIND', ' and to permit persons to whom the Software is\n        furnished to do so', ' and/or sell\n        copies of the Software', ' copy', ' distribute', ' free of charge', ' including without limitation the rights\n        to use', ' merge', ' modify', ' publish', ' subject to the following conditions:\n        \n        The above copyright notice and this permission notice shall be included in all\n        copies or substantial portions of the Software.\n        \n        THE SOFTWARE IS PROVIDED ""AS IS""', ' sublicense', ' to any person obtaining a copy\n        of this software and associated documentation files (the ""Software"")', ' to deal\n        in the Software without restriction', 'MIT License\n        \n        Copyright (c) 2014-2025 Beartype authors.\n        \n        Permission is hereby granted']","['Beartype']"
```
This ^ is a new dependency that doesn't get pulled in MacOs. Can we check why? Also, if we keep it, we need an override for this one to have a proper SPDX license.

```
26,28d24
< "extractcode","https://github.com/nexB/extractcode","['Apache-2.0']","['nexB. Inc. and others']"
< "extractcode-7z","https://github.com/nexB/scancode-plugins","['apache-2.0 AND lgpl-2.1 and unrar and brian-gladman-3-clause']","['nexB']"
< "extractcode-libarchive","https://github.com/nexB/scancode-plugins","['apache-2.0 AND bsd-simplified']","['nexB']"
```
These are ^ new dependencies that don't get pulled in MacOS.

```
43c39
< "markdown-it-py","https://github.com/executablebooks/markdown-it-py","['MIT']","['Copyright (c) 2020']"
---
> "markdown-it-py","https://github.com/executablebooks/markdown-it-py","['MIT']","['ExecutableBookProject']"
```
This dependency got the right copyright attribution in MacOs, but a wrong one in Linux. Can we check why? 

```
53c49
< "pip","https://github.com/pypa/pip","['MIT']","['Andrey Petrov and contributors', 'Corporation for National Research Initiatives', 'Eric Larson', 'INADA Naoki', 'Kim Davies and contributors', 'Python Software Foundation', 'Seth Michael Larson', 'Stephen Rosen', 'Stichting Mathematisch Centrum Amsterdam, The Netherlands', 'Taneli Hukkinen', 'The pip developers', 'The platformdirs developers', 'Thomas Kluyver', 'Tzu-ping Chung', 'Will McGugan']"
---
> "pip","https://github.com/pypa/pip","['MIT']","['Andrey Petrov and contributors', 'Corporation for National Research Initiatives', 'Eric Larson', 'INADA Naoki', 'Kim Davies and contributors', 'Python Software Foundation', 'Seth Michael Larson', 'Stephen Rosen', 'Stichting Mathematisch Centrum Amsterdam, The Netherlands', 'Taneli Hukkinen', 'The pip', 'The platformdirs', 'Thomas Kluyver', 'Tzu-ping Chung', 'Will McGugan']"
```
It looks like for MacOs, some of the attribution names get strip after two words (i.e. `The pip` vs `The pip developers`. Any clues?

```
70,71c66
< "scancode-toolkit","https://github.com/aboutcode-org/scancode-toolkit","['Apache-2.0 AND CC-BY-4.0 AND LicenseRef-scancode-other-permissive AND LicenseRef-scancode-other-copyleft']","['nexB. Inc. and others']"
< "semantic-version","https://github.com/rbarrois/python-semanticversion","['BSD']","['Raphaël Barrois']"
---
> "scancode-toolkit","https://github.com/nexB/scancode-toolkit","['Apache-2.0 AND CC-BY-4.0 AND LicenseRef-scancode-other-permissive AND LicenseRef-scancode-other-copyleft']","['nexB. Inc. and others']"
80d74
< "typecode-libmagic","https://github.com/nexB/scancode-plugins","['apache-2.0 AND bsd-simplified-darwin AND (bsd-simplified AND public-domain AND bsd-new AND isc AND (bsd-new OR gpl-1.0-plus) AND bsd-original)']","['nexB']"
83d76
< "uritools","https://github.com/tkem/uritools/","['MIT']","['Thomas Kemmer']"
```
These seem differences between Linux and MacOs. I think we can ignore these.

sdavtaker committed:
Adding colors to the diff in the CI test for ddla LICENSE-3rdparty check

sdavtaker committed:
Inversing the diff parameters in CI for readability

sdavtaker committed:
Renaming the comandline script

When running with debugpy, the name fo the package matching
the name of the cli script produced an error in identifying
the package names in the include path making a crash at the
time of loading the configuration.

sdavtaker committed:
Adjusting override rules and reference file

sdavtaker committed:
Saving artifact with output in integration test failure

sdavtaker commented:
- Added conditional path fot the integration test, that where it fails to validate the output of the tool against this same repository it will upload an artifact with the proposed output. 
- Updated overrides
- Using the artifact of a failed run result, I reviewed and uploaded a new version of the LICENSE-3rdparty.csv file

arapulido reviewed (approved):
LGTM

sdavtaker commented:
/merge

sdavtaker commented:
/remove
